### PR TITLE
KAFKA-10768 Make ByteBufferInputStream.read(byte[], int, int) to follow the contract

### PR DESCRIPTION
**Pull request by bertber**
I made a test for ByteBufferInputStream in the ByteBufferLogInputStreamTest.
First, I add a ByteBuffer that it's not empty to the ByteBufferInputStream, in order to verify it.
After that, I try to use ByteBufferInputStream's read function and check return value whether it's correct.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
@bertber Please take a look at origin PR (https://github.com/apache/kafka/pull/3752/files#diff-ae627decd5dd27d053a7a9b051860f60df6b89c8884e50df2f9ae0de6a4645e5R40). It would be better to fix the scenario of 'len==0'.

File: clients/src/test/java/org/apache/kafka/common/record/ByteBufferLogInputStreamTest.java [hunk 1]
```diff
@@ -120,4 +121,26 @@ public void iteratorRaisesOnTooLargeRecords() {
         logInputStream.nextBatch();
     }
 
+    @Test
+    public void testReadUnsignedIntFromInputStream() {
```
Comment (chia7712) [inline clients/src/test/java/org/apache/kafka/common/record/ByteBufferLogInputStreamTest.java]: Could you move the tests to ```ByteBufferInputStreamTest```? If there is no such test class, feel free to create new one.

**Comment by chia7712** [comment]
Is ```kafka.server.ControllerMutationQuotaTest.testPermissiveDeleteTopicsRequest``` related to this PR?

**Comment by bertber** [comment]
> Is `kafka.server.ControllerMutationQuotaTest.testPermissiveDeleteTopicsRequest` related to this PR?

No, I don't think it is related to this PR.

File: clients/src/test/java/org/apache/kafka/common/record/ByteBufferLogInputStreamTest.java [hunk 2]
```diff
@@ -119,5 +119,4 @@ public void iteratorRaisesOnTooLargeRecords() {
         assertNotNull(logInputStream.nextBatch());
         logInputStream.nextBatch();
     }
-
```
Comment (chia7712) [inline clients/src/test/java/org/apache/kafka/common/record/ByteBufferLogInputStreamTest.java]: please revert this change.

File: clients/src/test/java/org/apache/kafka/common/utils/ByteBufferInputStreamTest.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import static org.junit.Assert.assertEquals;
+
+import java.nio.ByteBuffer;
+
+import org.junit.Test;
+
+public class ByteBufferInputStreamTest {
+
+    @Test
+    public void testReadUnsignedIntFromInputStream() throws Exception {
```
Comment (chia7712) [inline clients/src/test/java/org/apache/kafka/common/utils/ByteBufferInputStreamTest.java]: ```throws Exception ``` is useless.

File: clients/src/test/java/org/apache/kafka/common/record/ByteBufferLogInputStreamTest.java (hunk 2 above)
Comment (bertber) [reply clients/src/test/java/org/apache/kafka/common/record/ByteBufferLogInputStreamTest.java]: This change have been reverted.

File: clients/src/test/java/org/apache/kafka/common/utils/ByteBufferInputStreamTest.java (hunk 3 above)
Comment (bertber) [reply clients/src/test/java/org/apache/kafka/common/utils/ByteBufferInputStreamTest.java]: It have been removed.

File: clients/src/test/java/org/apache/kafka/common/utils/ByteBufferInputStreamTest.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.utils;
+
+import static org.junit.Assert.assertEquals;
```
Comment (chia7712) [inline clients/src/test/java/org/apache/kafka/common/utils/ByteBufferInputStreamTest.java]: We have a pending PR offering a consistent import order (see https://github.com/apache/kafka/pull/8404#discussion_r535791013). Could you please follow the rules (although it is not merged)?

1. kafka, org.apache.kafka
1. com, net, org
1. java, javax

**Review by chia7712 — APPROVED** [review]
@bertber This patch LGTM. one small comment is left. Please take a look

File: clients/src/test/java/org/apache/kafka/common/utils/ByteBufferInputStreamTest.java (hunk 4 above)
Comment (bertber) [reply clients/src/test/java/org/apache/kafka/common/utils/ByteBufferInputStreamTest.java]: I have put the java behind the org
